src/main/java/Duke/InputParser.java

```diff
@@ -0,0 +1,96 @@
+package Duke;
+
+public class InputParser {
+
+    private String userCommand;
+    private String inputBuffer;
+
+    public InputParser() {
+        inputBuffer = "";
+        userCommand = "";
+    }
+
+    private void clear() {
+        inputBuffer = "";
+        userCommand = "";
+    }
+
+    private boolean isValidCommand() {
+        switch (userCommand) {
+            case ("list"): //Fallthrough
+            case ("mark"): //Fallthrough
+            case ("unmark"): //Fallthrough
+            case ("todo"): //Fallthrough
+            case ("deadline"): //Fallthrough
```

Comment:
Perhaps remove the comment `//Fallthrough` since it is redundant. You can consider adding one fallthrough comment at the end of the case. (I see this in a couple of places, so do look into that)